fhirr4/ballerina/src/main/resources/fhirservice/fhir_preprocessor.bal

```diff
@@ -216,6 +219,33 @@ public isolated class FHIRPreprocessor {
                 processResult = check processCommonSearchParameter(parameterDef, fhirResourceType, queryParam,
                                                                         self.apiConfig, self.searchParamConfigMap);
 
+            } else if queryParam.name == PAGE_QUERY_PARAM {
+                int? pageSize = self.apiConfig.pageSize;
+                if pageSize is int {
```

Comment:
We should do this in the init() since no need to do the check and processing for every request?